### PR TITLE
Update conventional-changelog dependency

### LINKED ISSUE #5672: fix(changelog): include refactor commits
The Changelog seems to omit any refactor commits, even those with breaking changes. For example: https://github.com/angular/angular/commit/5514dc19d9779fb6b0d7bbae8268dc21dd1dde39

^^^
That commit was not included

CC @robwormald

### PR DESCRIPTION
**Pull request by jeffbcross**
I...
- updated our conventional-changelog version to latest, 
- installed conventional-changelog-cli, 
- deleted our old changelog generation script, 
- and added a new npm script (`npm run gen-changelog`)

I tested that this fixes #5672 by creating this commit:

```
refactor(test): testing changeling generation do not submit

BREAKING CHANGE:

THIS IS BREAKING!!!
```

Then I ran `npm run gen-changelog`

Then I checked CHANGELOG.md, which contained this markdown (and prepended safely without destroying the rest of the changelog).

`<begin CHANGELOG snippet>`
### Code Refactoring
- **test:** testing changeling generation do not submit ([17775b2](https://github.com/angular/angular/commit/17775b2))
### BREAKING CHANGES
- test: THIS IS BREAKING!!!

`<end CHANGELOG snippet>`

TODO: 
- [ ] Update our release docs to reflect new changelog script

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -15,7 +15,8 @@
     "url": "https://github.com/angular/angular.git"
   },
   "scripts": {
-    "postinstall": "node tools/npm/copy-npm-shrinkwrap && webdriver-manager update"
+    "postinstall": "node tools/npm/copy-npm-shrinkwrap && webdriver-manager update",
+    "gen-changelog": "conventional-changelog -p angular -i CHANGELOG.md -s -r 1"
```
Comment (IgorMinar) [inline package.json]: can you please move this to gulp instead. in gulp, package.json and bash scripts :)

**Comment by IgorMinar** [comment]
the rest looks great. thank you

File: package.json (hunk 1 above)
Comment (jeffbcross) [inline package.json]: Sure, just a task called `changelog`?

**Comment by jeffbcross** [comment]
I updated the PR...
 * removed the conventional-changelog-cli dependency
 * removed `gen-changelog` npm script
 * added gulp-conventional-changelog dependency
 * added gulp task `changelog`
 * verified `CHANGELOG.md` output was same as previous test

**Comment by IgorMinar** [comment]
lgtm

**Comment by jeffbcross** [comment]
Merged at 0ad1215a92e16aa36f3969e6116f68f4aa854cba

**Pull request closed without merging**